testplan/runners/pools/base.py

```diff
@@ -305,16 +307,12 @@ def _loop(self):
         self._worker_monitor.daemon = True
         self._worker_monitor.start()
 
-        while (self.active
-               and self.status.tag is not self.status.STOPPING
-               and self.status.tag is not self.status.STOPPED):
-
+        while self.active and not self._exit_loop:
             msg = self._conn.accept()
             if msg:
                 try:
                     self.logger.debug('Received message from worker: %s.', msg)
-                    with self._pool_lock:
```

Comment:
Trying to remove all the pool lock

Reply:
Any particular reason this change was needed? We can try it, but needs testing to make sure the work thread and monitor thread won't interact badly.